### PR TITLE
Publish latest common release on PR

### PR DESCRIPTION
**Pull request by p3dr0rv**
https://identitydivision.visualstudio.com/Engineering/_build?definitionId=1464
This pipeline publish common every time a branch is merged to dev

Why this pipeline?
The previous setup is more complex and uses multiple components (GitHub actions, azure pipelines in IDDP, and extra branch ):
- A GitHub action merge dev into the continuous-delivery branch, increment the last patch number, and trigger an azure pipeline
- The pipeline build and publish common

### CODE REVIEW

**Comment by shahzaibj** [comment]
While you are here, can we also do it for the new [`LabApiUtilities` module](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/tree/dev/LabApiUtilities)? If it is too much work then we can leave it for now.

File: azure-pipelines/continous-delivery/common.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# File: azure-pipelines\continous-delivery\common.yml
+# Description: Assemble & publish common release to internal feed
+
+name: 0.$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger:
+- dev
```
Comment (shahzaibj) [inline azure-pipelines/continous-delivery/common.yml:7]: So this is doing the trick here? I assume having this here means the pipeline will run on every push to dev?

Comment (p3dr0rv) [reply azure-pipelines/continous-delivery/common.yml:10]: Correct

**Review by shahzaibj — APPROVED** [review]

File: LabApiUtilities/build.gradle [hunk 2]
```diff
@@ -76,3 +80,39 @@ afterEvaluate {
...
+        attributes('Implementation-Title': project.name,
+                'Implementation-Version': project.version)
+    }
+}
+
+publishing {
+    publications {
+        aar(MavenPublication) {
+            groupId 'com.microsoft.identity'
+            artifactId 'LabApiUtilities'
```
Comment (shahzaibj) [inline LabApiUtilities/build.gradle:95]: let's call it `lab-api-utilities` (I believe this is a more standard pattern for naming artifact ids)

Comment (p3dr0rv) [reply LabApiUtilities/build.gradle]: 88a34f4e372f037866173370dff119603a78f482 solved

**Comment by p3dr0rv** [comment]
Note: The scope of the PR increased during the review process

- remove the duplicated versioning file 
- add publish gradle task to LabApiUtilities

Result
https://identitydivision.visualstudio.com/Engineering/_build/results?buildId=843956&view=results

**Review by AdamBJohnsonx — APPROVED** [review]

File: azure-pipelines/continous-delivery/common.yml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      publishFolder: keyvault/build/libs
+# Lab API
+- job: labapi_phase
+  displayName: 'Lab API: Build, Test and Publish'
+  steps:
+  - template: ..\templates\steps\continous-delivery\assemble-publish-projversion.yml
+    parameters:
+      project: labapi
+      prjVersion: $(versionNumber)
+      publishFolder: labapi/build/libs
```
[the remark this answers is not in the document]
Comment (p3dr0rv) [reply azure-pipelines/continous-delivery/common.yml]: address comment here 3e977171bc858b5548a6b61bc7b01596ba3f7998

**Pull request merged**